File: extensions/nyno-uuidv4/command.js
// extensions/uuidv4-js/command.js

import crypto from "crypto";

export function nyno_uuidv4(args, context) {
  try {
    // Determine where to store the output
    const setName = context?.set_context ?? "prev";

    // Generate UUID v4
    const uuid =
      typeof crypto.randomUUID === "function"
        ? crypto.randomUUID()
        : "xxxxxxxx-xxxx-4xxx-yxxx-xxxxxxxxxxxx".replace(/[xy]/g, c => {
            const r = (Math.random() * 16) | 0;
            const v = c === "x" ? r : (r & 0x3) | 0x8;
            return v.toString(16);
          });

    // Store result in context
    context[setName] = uuid;

    return 0; // success
  } catch (error) {
    const setName = context?.set_context ?? "prev";
    context[setName + ".error"] = {
      errorMessage: error.message
    };
    return 1; // failure path
  }
}

File: extensions/nyno-array-to-item/command.py
# extensions/array_item_or_null/command.py

def nyno_array_to_item(args, context):
    # Determine output context key
    set_name = context.get('set_context', 'prev')

    # Check if args[0] exists
    if args and len(args) > 0:
        first_arg = args[0]
        # If it's a list, take the last item or None
        if isinstance(first_arg, list):
            value = first_arg[-1] if first_arg else None
        else:
            value = first_arg
    else:
        value = None

    context[set_name] = value

    # Return 1 if value is None, else 0
    return 1 if value is None else 0

